src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -170,6 +184,13 @@ public class SqlMutationEngine : IMutationEngine
                 throw _dabExceptionWithTransactionErrorMessage;
             }
 
+            if (!isReadPermissionConfigured)
+            {
+                throw new DataApiBuilderException(message: "The mutation operation was successful but the current user is unauthorized to view the response due to lack of read permissions",
+                                                  statusCode: HttpStatusCode.Forbidden,
+                                                  subStatusCode: DataApiBuilderException.SubStatusCodes.AuthorizationCheckFailed);
+            }
```

Comment:
in your pr description can you add a screenshot of the full GraphQL response before this change? by throwing an exception here, are we losing any details from the original error (before this change) and the more informative error message contents. e.g. are we missing the hotchocolate inserted error metadata that states which fields are error'd out?

Reply:
The new error message is now incorrect then compared to the original error message returned to the user for the following scenario:

Execute a read (or a mutation follow-up read) within the anonymous role context when the read role is only assigned to users in the authenticated role. The error the user should receive is `UNAUTHENTICATED` as referenced in the original error message. The authentication checks occur first prior to the authorization checks. With HotChocolate/GraphQL, the `@authorize` directive ensures that:

1. The user is authenticated
2. The user's role membership and `x-ms-api-role` header map to a role defined in the directive.

This PR's changes would only result in a valid error message for a follow-up read when both of these criteria hold true:

1. The user is authenticated
2. Both system roles `anonymous` and `authenticated` lack assigned read permissions.

Reply:
No, we get the same `UNAUTHENTICATED` error as before. 

Scenario: Anonymous has no read, Authenticated has read permission configured.

When a query is executed in the context of Anonymous role, the error message is as follows (with the changes in this PR):

![image](https://github.com/Azure/data-api-builder/assets/11196553/4f05e480-10f0-4982-a767-290647f04d68)

For the same query, the error message in version 0.9.7 is as follows:

![image](https://github.com/Azure/data-api-builder/assets/11196553/9441ff3b-8880-4f75-9e22-ff67a69c939e)